src/AutorestSwift/Models/Complex/Parameter.swift

```diff
@@ -66,6 +74,28 @@ class Parameter: Value, CustomDebugStringConvertible {
     public var debugDescription: String {
         return debugString() ?? description
     }
+
+    internal func belongsInSignature() -> Bool {
+        // We track body params in a special way, so always omit them generally from the signature.
+        // If they belong in the signature, they will be added in a way to ensure they come first.
+        guard paramLocation != .body else { return false }
+
+        // Default logic
```

Comment:
What does 'default logicl' here means?

Reply:
The special casing logic appears above, so this represents the default logic. I also changed it to make it hopefully more self-documenting, and split it between Parameter and VirtualParameter since they have different logic.